crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -1254,60 +1251,98 @@ impl CumulocityConverter {
+                registration.push(self.request_software_list(target));
+                Ok(registration)
+            }
         }
-        OperationStatus::Successful => {
-            let smartrest_set_operation =
-                SmartRestSetOperationToSuccessful::from_thin_edge_json(response)?.to_smartrest()?;
+    }
 
-            validate_and_publish_software_list(json_response, http_proxy, device_id).await?;
-            Ok(vec![Message::new(&topic, smartrest_set_operation)])
-        }
-        OperationStatus::Failed => {
-            let smartrest_set_operation =
-                SmartRestSetOperationToFailed::from_thin_edge_json(response)?.to_smartrest()?;
-            validate_and_publish_software_list(json_response, http_proxy, device_id).await?;
-            Ok(vec![Message::new(&topic, smartrest_set_operation)])
+    async fn publish_operation_status(
+        &self,
+        json_response: &str,
+    ) -> Result<Vec<Message>, CumulocityMapperError> {
+        let response = SoftwareUpdateResponse::from_json(json_response)?;
+        let topic = C8yTopic::SmartRestResponse.to_topic()?;
+        match response.status() {
+            OperationStatus::Executing => {
+                let smartrest_set_operation_status =
+                    SmartRestSetOperationToExecuting::from_thin_edge_json(response)?
+                        .to_smartrest()?;
+                Ok(vec![Message::new(&topic, smartrest_set_operation_status)])
+            }
+            OperationStatus::Successful => {
+                let smartrest_set_operation =
+                    SmartRestSetOperationToSuccessful::from_thin_edge_json(response)?
+                        .to_smartrest()?;
+
+                Ok(vec![
+                    Message::new(&topic, smartrest_set_operation),
+                    self.request_software_list(&EntityTopicId::default_main_device()), // FIXME: handle child devices
+                ])
+            }
+            OperationStatus::Failed => {
+                let smartrest_set_operation =
+                    SmartRestSetOperationToFailed::from_thin_edge_json(response)?.to_smartrest()?;
+                Ok(vec![
+                    Message::new(&topic, smartrest_set_operation),
+                    self.request_software_list(&EntityTopicId::default_main_device()), // FIXME: handle child devices
+                ])
+            }
         }
     }
-}
 
-async fn validate_and_publish_software_list(
-    payload: &str,
-    http_proxy: &mut C8YHttpProxy,
-    device_id: String,
-) -> Result<Vec<Message>, CumulocityMapperError> {
-    let response = &SoftwareListResponse::from_json(payload)?;
-
-    match response.status() {
-        OperationStatus::Successful => {
-            let c8y_software_list: C8yUpdateSoftwareListResponse = response.into();
-            http_proxy
-                .send_software_list_http(c8y_software_list, device_id)
-                .await?;
-        }
+    async fn validate_and_publish_software_list(
```

Comment:
note: It's not obvious what kind of validation does this function perform, so I'd like to see a doc-comment here, as this isn't a new function, it's not a blocker.

Reply:
Indeed, not a good name. Fixed with https://github.com/thin-edge/thin-edge.io/pull/2307/commits/4ef53919ee13e959d61a8129f8688865e9821036